server/fleet/hosts.go

```diff
@@ -555,16 +555,27 @@ func (h *Host) NeedsDEPEnrollment() bool {
 		h.IsDEPAssignedToFleet()
 }
 
-// IsElegibleForMicrosoftMDMEnrollment returns true if the host can be enrolled
-// in Fleet's Microsoft MDM (if Microsoft MDM was enabled).
-func (h *Host) IsElegibleForMicrosoftMDMEnrollment() bool {
+// IsEligibleForMicrosoftMDMEnrollment returns true if the host can be enrolled
+// in Fleet's Microsoft Windows MDM (if it was enabled).
+func (h *Host) IsEligibleForMicrosoftMDMEnrollment() bool {
 	return h.FleetPlatform() == "windows" &&
 		h.IsOsqueryEnrolled() &&
 		!h.MDMInfo.IsEnrolledInThirdPartyMDM() &&
 		!h.MDMInfo.IsFleetEnrolled() &&
 		(h.MDMInfo == nil || !h.MDMInfo.IsServer)
 }
 
+// IsEligibleForMicrosoftMDMUnenrollment returns true if the host must be
+// unenrolled from Fleet's Microsoft Windows MDM (if it MDM was disabled).
+func (h *Host) IsEligibleForMicrosoftMDMUnenrollment() bool {
+	return h.FleetPlatform() == "windows" &&
+		h.IsOsqueryEnrolled() &&
+		// TODO(mna): should that be only IsManualFleetEnrolled? Other types of
+		// enrollment will possibly require a different type of unenrollment?
```

Comment:
TBD if those conditions are correct for unenrollment.

Reply:
These conditions are OK. In the future, there will be a new enrollment type for Windows to handle Windows Autopilot enrollment (similar to DEP). The new enrollment will be another check in IsFleetEnrolled()